knowledge_share/views.py

```diff
@@ -20,15 +20,16 @@
 
 
 def _normalize_and_split_data(text):
-    # Normalize and remove empty spaces.
-    content = re.sub('[ ](?=[^\]]*?(?:\[|$))', '', text)
-    content = content.replace('][', '][').replace('],[', '][')
-
     # Remove first and last itens and split the string into a list.
-    content = content[1:-1].split('][')
-    if len(content) == 1:
-        return content[0], ''
-    return content
+    text = text.strip()
+    last_open_sqbra = text.rfind('[')
+    if text.endswith(']') and last_open_sqbra != -1:
+        categories_str = text[last_open_sqbra + 1:-1]
+        message = text[:last_open_sqbra]
+    else:
+        categories_str = ''
```

Comment:
You can put that before the `if`. The same to `message = text`. You don't need the `else`.